File: OrchidCore/src/main/java/com/eden/orchid/impl/tasks/DeployTask.java
package com.eden.orchid.impl.tasks;

import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.api.options.annotations.Description;
import com.eden.orchid.api.tasks.OrchidTask;
import com.eden.orchid.api.tasks.TaskService;
import com.google.inject.Provider;

import javax.inject.Inject;
import javax.inject.Named;

@Description("Run the main Orchid build process, publish the results, then exit.")
public final class DeployTask extends OrchidTask {

    private final Provider<OrchidContext> contextProvider;
    private final boolean dryDeploy;

    @Inject
    public DeployTask(Provider<OrchidContext> contextProvider, @Named("dryDeploy") Boolean dryDeploy) {
        super(100, "deploy", TaskService.TaskType.DEPLOY);
        this.contextProvider = contextProvider;
        this.dryDeploy = dryDeploy;
    }

    @Override
    public void run() {
        contextProvider.get().build();
        contextProvider.get().deploy(dryDeploy);
    }

}

File: OrchidCore/src/main/kotlin/com/eden/orchid/impl/tasks/DeployTask.kt
package com.eden.orchid.impl.tasks

import com.eden.orchid.api.OrchidContext
import com.eden.orchid.api.options.annotations.Description
import com.eden.orchid.api.tasks.OrchidTask
import com.eden.orchid.api.tasks.TaskService
import javax.inject.Inject
import javax.inject.Named

@Description("Run the main Orchid build process, publish the results, then exit.")
class DeployTask
@Inject
constructor(
    @Named("dryDeploy") private val dryDeploy: Boolean
) : OrchidTask(100, "deploy", TaskService.TaskType.DEPLOY) {

    override fun run(context: OrchidContext) {
        context.build()
        val success = context.deploy(dryDeploy)

        if (!success) {
            throw Exception("deployment failed, check logs for details")
        }
    }

}
